commons/src/main/scala/com/lynbrookrobotics/potassium/commons/drivetrain/UnicycleCoreTasks.scala

```diff
@@ -175,4 +178,53 @@ trait UnicycleCoreTasks {
       drive.resetToDefault()
     }
   }
+
+  class CorrectOffsetWithLatency(timestampedOffset: Signal[(Angle, Time)], tolerance: Angle)
+    (implicit drive: Drivetrain,
+      hardware: DrivetrainHardware,
+      props: Signal[DrivetrainProperties]) extends FiniteTask {
+
+    val positionSlide = hardware.turnPosition.toPeriodic.slidingWithTime(
+      20, hardware.turnPosition.get
+    ).toPollingSignal(Milliseconds(5))(drive.clock)
+
+    override def onStart(): Unit = {
+      val targetAbsolute = timestampedOffset.map{ offsetWithTime =>
```

Comment:
The logic for control should go into a controller method, so that we can apply different variations (such as using an open-loop lower level control).